cleaner/mod.ts

```diff
@@ -0,0 +1,117 @@
+import { bold, CommandHandler, fmt, Module, updateMessage } from "$xor";
+import { Api } from "$grm";
+
+const I_PTO = "purge__purge_to";
+const I_PFROM = "purge__purge_from";
+
+const cleaner: Module = {
+  name: "cleaner",
+  handlers: [
+    new CommandHandler("purge", async ({ event, client, args }) => {
+      let purgeTo = Number(sessionStorage.getItem(I_PTO));
+      const purgeFrom = Number(sessionStorage.getItem(I_PFROM));
+      if (!purgeTo) {
+        const reply = await event.message.getReplyMessage();
+        if (!reply) {
+          await updateMessage(
+            event,
+            "Reply a message to purge to or mark with pto (and pfrom).",
+          );
+          return;
+        }
+        purgeTo = reply.id;
+      }
+      sessionStorage.removeItem(I_PTO);
+      sessionStorage.removeItem(I_PFROM);
+      let messages = new Array<Api.Message>();
+      const self = args[0]?.startsWith("s");
+      const all = args[1] == "ALL" && self;
+      for await (
+        const message of client.iterMessages(event.chatId, {
+          minId: all ? undefined : purgeTo - 1,
+          maxId: all ? undefined : purgeFrom == 0 ? undefined : purgeFrom + 1,
```

Comment:
Just curious if it supports more than 100 messages in single run, because in clients they limit it to 100 selection maximum

Reply:
Sorry, my bad, needed to check full